packages/nodes-base/nodes/SIGNL4/GenericFunctions.ts

```diff
@@ -0,0 +1,64 @@
+import {
+	IExecuteFunctions,
+	IHookFunctions,
+} from 'n8n-core';
+
+import {
+	ICredentialDataDecryptedObject,
+	IDataObject,
+} from 'n8n-workflow';
+
+/**
+ * Make an API request to MSG91
+ *
+ * @param {IHookFunctions | IExecuteFunctions} this
+ * @param {object} message
+ * @returns {Promise<any>}
+ */
+export async function SIGNL4ApiRequest(this: IHookFunctions | IExecuteFunctions, message: IDataObject): Promise<any> { // tslint:disable-line:no-any
+	const credentials = this.getCredentials('SIGNL4Api');
+	if (credentials === undefined) {
+		throw new Error('No credentials got returned!');
+	}
+	
+	const teamsecret = credentials!.teamSecret as string;
+
+	var response = null;
+	try {
+		response = await this.helpers.request({
+			headers: {'Content-Type': 'application/json'},
+			method: 'POST',
+			body: message,
+			qs: {},
+			uri: `https://connect.signl4.com/webhook/${teamsecret}`,
+			json: true
+		});
+	}
+	catch (error) {
+		
+		throw new Error(`I am behind. Status: ${JSON.stringify(teamsecret)}`);
```

Comment:
Do not understand this part. What does it mean "I am behind...?" and why does it output the "secret" access token?

Why is there any more code afterward that would never get called?